source/test/TestApp/TestUtils.cpp

```diff
@@ -3,8 +3,44 @@
 #include "TestAppPch.h"
 #include "TestUtils.h"
 
+#include <random>
+#include <sstream>
+
 namespace PlayFabUnit
 {
+    
+    std::string GenerateUuidV4() {
+        std::random_device              rd;
+        std::mt19937                    gen(rd());
+        std::uniform_int_distribution<> dis(0, 15);
+        std::uniform_int_distribution<> dis2(8, 11);
+
+        std::stringstream ss;
+        int i;
+        ss << std::hex;
+        for (i = 0; i < 8; i++) {
+            ss << dis(gen);
```

Comment:
By calling the same randomizer 4 times in sequence, you're not actually generating guid's as randomly as guids are supposed to be.
Guids have 128 bits of randomness, but in your case, your method only has 32.
Repeatedly calling dis() sequentially won't be 128 bits of randomness, but only 32, because only the first number is "random", since it was seeded by the time. Every number after the first one is predicatable, and based on only the first number.

Meaning that this GUID generator only has 4E9 (4 billion) possible outputs rather than the intended 3E38 (300 undecillion or 300 billion billion billion billion) outputs.

In short, it generates a number that is in guid format, but it's not actually a reliable guid in terms of being super unique. In fact, it's extremely likely to be not unique at all, and if our customers rely on these values being unique, they'll very quickly be disappointed.

Reply:
renamed to not call it a guid.